sdks/java/io/rabbitmq/src/test/java/org/apache/beam/sdk/io/rabbitmq/RabbitMqIOTest.java

```diff
@@ -98,60 +98,65 @@ public static void beforeClass() throws Exception {
     attributes.put(SystemConfig.DEFAULT_QPID_WORK_DIR, temporaryFolder.newFolder().toString());
 
     launcher.startup(attributes);
+
+    uri = "amqp://guest:guest@localhost:" + port;
+
+    connectionHandler = new ConnectionHandler(uri);
   }
 
   @AfterClass
   public static void afterClass() {
-    launcher.shutdown();
     if (defaultPort != null) {
       System.setProperty("qpid.amqp_port", defaultPort);
     } else {
       System.clearProperty("qpid.amqp_port");
     }
+    launcher.shutdown();
+    try {
+      connectionHandler.close();
+    } catch (IOException e) {
+      /* ignored */
+    }
   }
 
-  @Test
-  public void testReadQueue() throws Exception {
+  @Test(timeout = ONE_MINUTE_MS)
+  public void testReadDefaultExchangeExistingQueue() throws Exception {
```

Comment:
This is the only test I've as-of-yet rewritten and have seen it pass just to prove basic functionality.